README.md

```diff
@@ -1,2 +1,140 @@
+
+A relay is needed in situations where nodes are behind NAT, reverse proxies, firewalls and/or simply don't support the same transports (e.g. go-libp2p vs. browser-libp2p). The circuit relay protocol exists to overcome those scenarios. Nodes with the `auto-relay` feature enabled can automatically bind themselves on a relay to listen for connections on their behalf.
+
+You can read more in its [SPEC](https://github.com/libp2p/specs/tree/master/relay).
+
+## Usage
+
+### Install
+
+```bash
+> npm install --global libp2p-relay-server
+```
+
+Now you can use the cli command `libp2p-relay-server` to spawn an out of the box libp2p relay server.
+
+### CLI
+
+After installing the relay server, you can use its binary. It accepts several arguments: `--peerId`, `--listenMultiaddrs`, `--announceMultiaddrs`, `--metricsMultiaddr`, `--disableMetrics`, `--disablePubsubDiscovery` and `--discoveryTopics`.
+
+```sh
+libp2p-relay-server [--peerId <jsonFilePath>] [--listenMultiaddrs <ma> ... <ma>] [--announceMultiaddrs <ma> ... <ma>] [--metricsMultiaddr <ma>] [--disableMetrics] [--disablePubsubDiscovery] [--discoveryTopics <topic> ... <topic>]
+```
+
+#### PeerId
+
+You can create a [PeerId](https://github.com/libp2p/js-peer-id) via its [CLI](https://github.com/libp2p/js-peer-id#cli). 
+
+```sh
+libp2p-relay-server --peerId id.json
+```
+
+#### Multiaddrs
+
+You can specify the libp2p rendezvous server listen and announce multiaddrs. This server is configured with [libp2p-tcp](https://github.com/libp2p/js-libp2p-tcp) and [libp2p-websockets](https://github.com/libp2p/js-libp2p-websockets) and addresses with this transports should be used. It can always be modified via the API.
+
+```sh
+libp2p-relay-server --peerId id.json --listenMultiaddrs '/ip4/127.0.0.1/tcp/15002/ws' '/ip4/127.0.0.1/tcp/8000' --announceMultiaddrs '/dns4/test.io/tcp/443/wss/p2p/12D3KooWAuEpJKhCAfNcHycKcZCv9Qy69utLAJ3MobjKpsoKbrGA' '/dns6/test.io/tcp/443/wss/p2p/12D3KooWAuEpJKhCAfNcHycKcZCv9Qy69utLAJ3MobjKpsoKbrGA'
+```
+
+By default it listens on `/ip4/127.0.0.1/tcp/15002/ws` and has no announce multiaddrs specified.
+
+#### Metrics
+
+Metrics are enabled by default on `/ip4/127.0.0.1/tcp/8003` via Prometheus. This address can also be modified with:
+
+```sh
+libp2p-relay-server --metricsMultiaddr '/ip4/127.0.0.1/tcp/8000'
+```
+
+Moreover, metrics can also be disabled with:
+
+```sh
+libp2p-relay-server --disableMetrics
+```
+
+#### Discovery
+
+A discovery module [libp2p/js-libp2p-pubsub-peer-discovery](https://github.com/libp2p/js-libp2p-pubsub-peer-discovery) is configured and enabled by default. It can be disabled with:
+
+```sh
+libp2p-relay-server --disablePubsubDiscovery
+```
+
+This discovery module uses `_peer-discovery._p2p._pubsub` as a topic for enabling peer discovery. However, other topics can be used with:
+
+```sh
+libp2p-relay-server --discoveryTopics '_peer-discovery._app1._pubsub' '_peer_discovery._app2._pubsub'
+```
```

Comment:
I'd just leave configuring this out, keep it simple. It binds to the global space out of the box, so just avoid people doing the wrong thing. IMO if people need to configure this in depth they can use the config as a template. The point of this repo is basically to get people up and running quickly, so we should just reduce the noise. 

What's the bare minimum I need to do to get this running? The readme should focus on that. I think a consistent PeerId and Announce addresses are the only thing I should actually need to set. I'd prioritize and separate the need to have, from the optional stuff.

Reply:
I divided them in two now. The required/recommended is in the usage section: `peerId`, `multiaddrs` and `ssl`

Then a new configuration section created for fine tuning of the image